qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,601 @@
+
+        self._min_freq: float
+        self._max_freq: float
+        self._additional_wait: float = 1
+
+        self.add_parameter(
+            name="mode",
+            get_cmd=":INSTrument:SELect?",
+            set_cmd=":INSTrument:SELect {}",
+            vals=Enum(*self._available_modes()),
+            docstring="Allows setting of different modes present and licensed "
+                      "for the instrument."
+        )
+
+        self.add_parameter(
+            name="measurement",
+            get_cmd=":CONFigure?",
+            set_cmd=":CONFigure:{}",
+            vals=Enum("SAN", "LPL"),
+            docstring="Sets measurement type from among the available "
+                      "measurement types."
+        )
+
+        self.add_parameter(
+            name="cont_meas",
+            initial_value=False,
+            get_cmd=":INITiate:CONTinuous?",
+            set_cmd=self._enable_cont_meas,
+            val_mapping=create_on_off_val_mapping(on_val="ON", off_val="OFF"),
+            docstring="Enables or disables continuous measurement."
+        )
+
+        self.add_parameter(
+            name="format",
+            get_cmd=":FORMat:TRACe:DATA?",
+            set_cmd=":FORMat:TRACe:DATA {}",
+            val_mapping={
+                "ascii": "ASCii",
+                "int32": "INTeger,32",
+                "real32": "REAL,32",
+                "real64": "REAL,64"
+            },
+            docstring="Sets up format of data received"
+        )
+
+        if "SA" in self._available_modes():
+            sa_mode = SpectrumAnalyzerMode(self, name="sa")
+            self.add_submodule("sa", sa_mode)
+        else:
+            self.log.info("Spectrum Analyzer mode is not available on this "
+                          "instrument.")
+
+        if "PNOISE" in self._available_modes():
+            pnoise_mode = PhaseNoiseMode(self, name="pn")
+            self.add_submodule("pn", pnoise_mode)
+        else:
+            self.log.info("Phase Noise mode is not available on this "
+                          "instrument.")
+        self.connect_message()
+
+    def _available_modes(self) -> Tuple[str, ...]:
+        """
+        Returns present and licensed modes for the instrument.
+        """
+        available_modes = self.ask(":INSTrument:CATalog?")
+        av_modes = available_modes[1:-1].split(',')
+        modes: Tuple[str, ...] = ()
+        for i, _ in enumerate(av_modes):
+            if i == 0:
+                modes = modes + (av_modes[i].split(' ')[0],)
+            else:
+                modes = modes + (av_modes[i].split(' ')[1],)
+        return modes
+
+    def _available_meas(self) -> Tuple[str, ...]:
+        """
+        Gives available measurement with a given mode for the instrument
+        """
+        available_meas = self.ask(":CONFigure:CATalog?")
+        av_meas = available_meas[1:-1].split(',')
+        meas: Tuple[str, ...] = ()
+        for i, _ in enumerate(av_meas):
```

Comment:
This is a slightly odd way to use enumerate. I would suggest never to use enumerate if you are actually not using the thing that you are enumerating over. 

So I suggest either change `for i, _ in enumerate(av_meas)` back to for i in range(len(av_meas))`
or better actually use the thing that you are enumerating over.

```python
measurements = []
for i, meas in enumerate(av_meas):
    if i == 0:
        measurements = measurements + (meas,)
   else:
       measurements = measurements + (meas[1:],)
return tuple(measurements)
```